### PR TITLE
enh(secu): add rule for max session duration - for master

### PR DESCRIPTION
**Pull request by sc979**
# Pull Request Template

## Description

Add a rule in the administration form to require a session value lesser or equal to 120min.
Update the user's saved session value in the DB if greater to 120min

image

**Fixes** # (secu)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [x] 19.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/include/Administration/parameters/general/form.php [hunk 1]
```diff
@@ -62,8 +67,8 @@
 /*
  * Style
  */
-$attrsText        = array("size"=>"40");
-$attrsText2        = array("size"=>"5");
+$attrsText = array("size"=>"40");
```
Comment (kduret) [inline www/include/Administration/parameters/general/form.php]:
```suggestion
$attrsText = array("size" => "40");
```

File: www/include/Administration/parameters/general/form.php [hunk 2]
```diff
@@ -62,8 +67,8 @@
...
+$attrsText2 = array("size"=>"5");
```
Comment (kduret) [inline www/include/Administration/parameters/general/form.php]:
```suggestion
$attrsText2 = array("size" => "5");
```

File: www/include/Administration/parameters/DB-Func.php [hunk 3]
```diff
@@ -113,6 +113,18 @@ function is_writable_file_if_exist($path = null)
...
 }
 
+/**
+ * rule to check the session duration value chosen by the user
+ * @param int $value
+ * @param int $valueMax
+ *
+ * @return bool
+ */
+function is_session_duration_valid(int $value = null)
```
Comment (kduret) [inline www/include/Administration/parameters/DB-Func.php]: can you use camelcase ?

File: www/include/Administration/parameters/general/help.php [hunk 4]
```diff
@@ -37,7 +37,10 @@
  * Sessions Properties
  */
 
-$help['tip_sessions_expiration_time'] = dgettext('help', 'Life duration of sessions.');
+$help['tip_sessions_expiration_time'] = dgettext(
+    'help',
+    'Life duration of sessions. The value in minutes cannot be greater than the value set in your php centreon.ini file.'
```
Comment (ganoze) [inline www/include/Administration/parameters/general/help.php]: It might be worth mentioning the exact PHP parameter (_session.gc_maxlifetime_).

Comment (sc979) [reply www/include/Administration/parameters/general/help.php]: There's only four values. And only the admin of the server can modify those.
But why not. I'll modify the 15 files :'(

Comment (sc979) [reply www/include/Administration/parameters/general/help.php]: done ;)

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 5]
```diff
@@ -14839,3 +14839,6 @@ msgstr "Ajouter des Remote Server additonnels"
 
 msgid "Locked elements"
 msgstr "Eléments verrouillés"
+
+msgid "This value needs to be an integer lesser than"
+msgstr "Cette valeur doit être inférieur à"
```
Comment (kduret) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "Cette valeur doit être inférieure à"
```

File: www/install/php/Update-19.10.0-rc.2.php [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    if ($sessionValue > 120) {
+        $pearDB->query(
+            'UPDATE `options` SET `value` = "120"
+            WHERE `key` = "session_expire"'
+        );
+    }
+} catch (\PDOException $e) {
+    $centreonLog->insertLog(
+        2,
+        "UPGRADE : 19.04.5 Unable to modify session expiration value"
```
Comment (kduret) [inline www/install/php/Update-19.10.0-rc.2.php]:
```suggestion
        "UPGRADE : 19.10.0 Unable to modify session expiration value"
```

File: lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 7]
```diff
@@ -14451,3 +14451,7 @@ msgstr "Para usar Remote Server adicionais, um Remote Server mestre deve ser sel
 
 msgid "Attach additional Remote Servers"
 msgstr "Anexar Remote Server adicionais"
+
+
```
Comment (sc979) [inline lang/pt_BR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
```

File: lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 8]
```diff
@@ -14452,3 +14452,7 @@ msgstr "Para usar Remote Server adicionais, um Remote Server mestre deve ser sel
 
 msgid "Attach additional Remote Servers"
 msgstr "Anexar Remote Server adicionais"
+
+
```
Comment (sc979) [inline lang/pt_PT.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
```

**Pull request merged**